packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte

```diff
@@ -0,0 +1,75 @@
+<script lang="ts">
+import SettingsPage from './SettingsPage.svelte';
+import EngineIcon from '../ui/EngineIcon.svelte';
+import EmptyScreen from '../ui/EmptyScreen.svelte';
+import { onMount } from 'svelte';
+import type { KubeContextUI } from '../kube/KubeContextUI';
+import { getKubeUIContexts } from '../kube/KubeContextUI';
+import Link from '../ui/Link.svelte';
+
+let contexts: KubeContextUI[] = [];
+
+$: contexts;
+
+onMount(async () => {
+  // Retrieve both the contexts and clusters
+  let k8sContexts = await window.kubernetesGetContexts();
+  let k8sClusters = await window.kubernetesGetClusters();
+
+  // Convert them to KubeContextUI so we can safely render them.
+  contexts = getKubeUIContexts(k8sContexts, k8sClusters);
+});
+</script>
```

Comment:
From what I see we collect only the contexts/clusters when we display the page

so if I create a cluster, delete, etc while being on the page it won't be refreshed.

I think we need to use the store pattern and events to correctly have the UI being refreshed from changes and not from when we display the UI

Reply:
Yes, we will be adding this as part of the second phase since we need to add actions such as delete / create which would be in a store: https://github.com/containers/podman-desktop/issues/4416